beacon-chain/execution/engine_client.go

```diff
@@ -714,7 +723,7 @@ func fullPayloadFromExecutionBlock(
 		txs[i] = txBin
 	}
 
-	switch block.Version {
+	switch blockVersion {
```

Comment:
why doesn't block.version work?

Reply:
The intent of this change was to replace code that relied on heuristics inside the execution block unmarshal function to guess the version of the block, because this code was buggy, which led to the original type to be mutated. Since the caller of this function knows the original blinded type, I thought it made more sense to let the caller specify the type to use, and avoid the buggy guessing game entirely.